### PR TITLE
OBPIH-2595: Ability to import PO items from a template

### PR DESCRIPTION
**Pull request by awalkowiak**

### CODE REVIEW

File: grails-app/controllers/org/pih/warehouse/order/OrderController.groovy [hunk 1]
```diff
@@ -739,17 +739,23 @@ class OrderController {
             redirect(action: "list")
         } else {
             def date = new Date()
-            response.setHeader("Content-disposition", "attachment; filename='\"${orderInstance.orderNumber}-${date.format("MM-dd-yyyy")}.csv\"")
+            response.setHeader("Content-disposition", "attachment; filename=${orderInstance.orderNumber}-${date.format("MM-dd-yyyy")}.csv")
```
Comment (jmiranda) [inline grails-app/controllers/org/pih/warehouse/order/OrderController.groovy:742]: You can get rid of that first single quote, but I think the escaped double quote is there for a reason. I believe we had some issues with Firefox.

File: grails-app/services/org/pih/warehouse/order/OrderService.groovy [hunk 2]
```diff
@@ -534,20 +537,72 @@ class OrderService {
...
+                        }
+                        UnitOfMeasure uom = uomParts.length > 1 ? UnitOfMeasure.findByName(uomParts[0]) : null
+                        BigDecimal qtyPerUom = uomParts.length > 1 ? BigDecimal.valueOf(Double.valueOf(uomParts[1])) : null
+                        orderItem.quantityUom = uom
+                        orderItem.quantityPerUom = qtyPerUom
+                    }
+
+                    orderItem.quantity = Integer.valueOf(quantity)
+                    orderItem.unitPrice = BigDecimal.valueOf(Integer.valueOf(unitPrice))
+                    orderItem.recipient = recipientNames.length > 1 ?
```
Comment (jmiranda) [inline grails-app/services/org/pih/warehouse/order/OrderService.groovy:600]: Consider reusing/improving PersonDataService.createOrUpdatePerson(Map)

Comment (jmiranda) [reply grails-app/services/org/pih/warehouse/order/OrderService.groovy:600]: In fact, search the code by "findByFirstNameAndLastName" and there should be three references to similar code. Let's try to refactor it so we're reusing the same code in all three places (if it makes sense).

File: grails-app/services/org/pih/warehouse/order/OrderService.groovy [hunk 3]
```diff
@@ -572,14 +627,28 @@ class OrderService {
      * @param inputStream
      * @return
      */
-    List parseOrderItems(String text) {
+    List parseOrderItems(InputStream inputStream) {
```
Comment (jmiranda) [inline grails-app/services/org/pih/warehouse/order/OrderService.groovy:630]: Why did you change the signature here? Why not just use the following?
```
orderService.parseOrderItems(multipartFile.inputStream.text)
```

File: grails-app/services/org/pih/warehouse/order/OrderService.groovy (hunk 2 above)
Comment (awalkowiak) [reply grails-app/services/org/pih/warehouse/order/OrderService.groovy:600]: Ok, will do

File: grails-app/controllers/org/pih/warehouse/order/OrderController.groovy (hunk 1 above)
Comment (awalkowiak) [reply grails-app/controllers/org/pih/warehouse/order/OrderController.groovy:742]: Ok, will do

File: grails-app/services/org/pih/warehouse/order/OrderService.groovy (hunk 3 above)
Comment (awalkowiak) [reply grails-app/services/org/pih/warehouse/order/OrderService.groovy:630]: I think I had some kind of issues here, but I'll check  again

File: grails-app/services/org/pih/warehouse/order/OrderService.groovy (hunk 2 above)
Comment (awalkowiak) [reply grails-app/services/org/pih/warehouse/order/OrderService.groovy:600]: Hmm, I just noticed, that in requirements from this ticket there is said that if the person is not found, it should not create a new one. The code you mentioned is creating a new Person if they can't find it by first and last name. I'll just refactor the code using findByFirstNameAndLastName.
